lib/Doctrine/ORM/Query/Parser.php

```diff
@@ -949,29 +954,43 @@ public function AliasIdentificationVariable()
     }
 
     /**
-     * AbstractSchemaName ::= identifier
+     * AbstractSchemaName ::= fully_qualified_name | aliased_name | identifier
      *
      * @return string
      */
     public function AbstractSchemaName()
     {
-        $this->match(Lexer::T_IDENTIFIER);
-
-        $schemaName = ltrim($this->lexer->token['value'], '\\');
-
-        if (strrpos($schemaName, ':') !== false) {
-            list($namespaceAlias, $simpleClassName) = explode(':', $schemaName);
-
-            $schemaName = $this->em->getConfiguration()->getEntityNamespace($namespaceAlias) . '\\' . $simpleClassName;
+        if ($this->lexer->isNextToken(Lexer::T_FULLY_QUALIFIED_NAME)) {
```

Comment:
@guilhermeblanco Don't worry, I am glad you find the time to review this. This is definetly too big for my brain alone.

This should not be a showstopper, you guys just have to decide how it should work.